test/api/ticket/api_test.go

```diff
@@ -0,0 +1,164 @@
+package ticket
+
+import (
+	"testing"
+
+	"github.com/konveyor/tackle2-hub/api"
+	TrackerSamples "github.com/konveyor/tackle2-hub/test/api/tracker"
+	"github.com/konveyor/tackle2-hub/test/assert"
+)
+
+func TestTicketCRUD(t *testing.T) {
+	for _, r := range Samples {
+		t.Run("Ticket "+r.Kind+" CRUD", func(t *testing.T) {
+
+			// Create a sample Application for the ticket.
+			app := api.Application{
+				Name: r.Application.Name,
+			}
+			assert.Must(t, Application.Create(&app))
+
+			createdIdentities := []api.Identity{}
+			createdTrackers := []api.Tracker{}
+			for _, tracker := range TrackerSamples.Samples {
+				// Create a sample identity for the tracker
+				identity := api.Identity{
+					Name: tracker.Identity.Name,
+					Kind: tracker.Kind,
+				}
+				assert.Must(t, Identity.Create(&identity))
+				createdIdentities = append(createdIdentities, identity)
+				assert.Must(t, Tracker.Create(&tracker))
+				createdTrackers = append(createdTrackers, tracker)
+			}
+
+			// Create a sample ticket
+			assert.Must(t, Ticket.Create(&r))
+
+			// Get.
+			got, err := Ticket.Get(r.ID)
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+
+			// Compare got values with expected values.
+			AssertEqualTickets(t, got, r)
+
+			// Delete ticket and its related resources.
+			assert.Must(t, Ticket.Delete(r.ID))
+			for _, tracker := range createdTrackers {
+				assert.Must(t, Tracker.Delete(tracker.ID))
+			}
+			for _, identity := range createdIdentities {
+				assert.Must(t, Identity.Delete(identity.ID))
+			}
+			assert.Must(t, Application.Delete(app.ID))
+
+			// Check if the Ticket is present even after deletion or not.
+			_, err = Ticket.Get(r.ID)
+			if err == nil {
+				t.Errorf("Resource exits, but should be deleted: %v", r)
+			}
+		})
+	}
+}
+
+func TestTicketList(t *testing.T) {
+	for _, r := range Samples {
+
+		createdTickets := []api.Ticket{}
+		// Create a sample Application for the ticket.
+		app := api.Application{
+			Name: r.Application.Name,
+		}
+		assert.Must(t, Application.Create(&app))
+
+		createdIdentities := []api.Identity{}
+		createdTrackers := []api.Tracker{}
+		for _, tracker := range TrackerSamples.Samples {
+			// Create a sample identity for the tracker
+			identity := api.Identity{
+				Name: tracker.Identity.Name,
+				Kind: tracker.Kind,
+			}
+			assert.Must(t, Identity.Create(&identity))
+			createdIdentities = append(createdIdentities, identity)
+			assert.Must(t, Tracker.Create(&tracker))
+			createdTrackers = append(createdTrackers, tracker)
+		}
+
+		// Create a sample ticket
+		assert.Must(t, Ticket.Create(&r))
+		createdTickets = append(createdTickets, r)
+
+		// List Tickets.
+		got, err := Ticket.List()
```

Comment:
There is no filtering parameter (like Application or Tracker ID), the filter is on API side in List action. Unless filter parameters are sent to API, I'm not sure how the filtering is tested.